### ISSUE TITLE
BatchNorm in eval() model causes validation errors

### ISSUE THREAD

**Issue by sssruhan1** [bearpaw/pytorch-pose#4]
Hi,

When I try to train hg8, it is giving me very large validation error (looks almost like it is not working), while in training mode, the loss is very small ( and looks working very well). I did some googling, and find out BatchNorm layer behaves differently in eval() mode and train() mode. 

So I adjusted the code 
in dataset/mpii.py __getItem__ function, where color_normalization should only be used for training as a pre-processing mode. 

After adjusting it worked well.

**Comment by bearpaw** [comment]
@sssruhan1 Hi. BatchNorm uses global (fixed) statistics during testing time. I cannot see the reason that `color_normalization` should not be used during testing time.

**Comment by sssruhan1** [comment]
Yeah, I find out when clamp the image for validation inside dataset/mpii, the result is good for my training. Oddly, when I use the given trained model (downloaded hg4 model for mpii), clamp was not needed. I am still not too sure what I have done wrong. Since I just used the given arguments for example/mpii.py .

**Comment by weigq** [comment]
@sssruhan1  have you solved it? i have the same problem here,

**Comment by xiaoyong** [comment]
After mean subtraction performed in `color_normalization`, the pixel values range from -mean to 1-mean instead of [0, 1]. So clamping should be applied **before** instead of after normalization.

BTW, in line
https://github.com/bearpaw/pytorch-pose/blob/8e46155fc96decff3f15b3c91f676fc82be736f4/pose/utils/imutils.py#L24

The trailing division by 255 is unnecessary since it is already done in `im_to_torch`.

**Comment by bearpaw** [comment]
@xiaoyong Thanks a lot! This is definitely a bug! 

Actually, I found there are some problems in `color_normalization`, but could not figure it out. I just delete `color_normalization` in the last push. I will fix the bug and update the code. Thank you again!

**Comment by bearpaw** [comment]
The bug caused the problem has been fixed.
